### PR TITLE
Wait for PG, don't ask password on detach

### PR DESCRIPTION
**Pull request by sdressler**

### CODE REVIEW

File: node-control.sh [hunk 1]
```diff
@@ -8,6 +8,29 @@ function print_help {
...
+        if [ $? -eq 0 ]; then
+            PSQL_UP=1
+            break
+        fi
+        sleep 15
+    done
+
+    if [ $PSQL_UP -ne 1 ]; then
+        echo "PSQL did not start within time. Aborting."
+        exit -1
```
Comment (lmcarril) [inline node-control.sh]: in bash, exit codes are positive numbers

File: node-control.sh [hunk 2]
```diff
@@ -8,6 +8,29 @@ function print_help {
   echo "Usage: $0 {attach|detach} <node-number>"
 }
 
+function wait_for_pg {
+    set +e
+    PG_HOST=$1
+
+    PSQL_UP=0
+    for i in {0..60}; do
+        pg_isready -h $PG_HOST
```
Comment (lmcarril) [inline node-control.sh:16]: another way to not have to disable the errors is to capture the return code:
xyz && retcode=$? || true

**Review by lmcarril — APPROVED** [review]

File: node-control.sh (hunk 1 above)
Comment (sdressler) [reply node-control.sh]: Fixed.

File: node-control.sh (hunk 2 above)
Comment (sdressler) [reply node-control.sh:16]: Was a leftover from another script which uses `set -e`. Removed.

**Pull request merged**